plugins/inputs/t128_lte/t128_lte.go

```diff
@@ -0,0 +1,384 @@
+package t128_lte
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+var serviceStatusMap = map[string]string{
+	"not-registered-searching": "Unable to Register",
+	"no-service":               "No Service Available",
+	"authentication-failed":    "Authentication Failure",
+	"plmn-not-allowed":         "Forbidden PLMN",
+}
+
+var sampleConfig = `
+[[inputs.t128_lte]]
+## Required. A name for the collector which will be used as the measurement name of the produced data.
+# collector_name = "lte-state"
+`
+
+type globalInitConfig struct {
+	Init struct {
+		RouterName string `json:"routerName"`
+	} `json:"init"`
+}
+
+type localInitConfig struct {
+	Init struct {
+		Id string `json:"id"`
+	} `json:"init"`
+}
+
+type nodes struct {
+	Config struct {
+		Authority struct {
+			Router []struct {
+				Name string `json:"name"`
+				Node []struct {
+					Name string `json:"name"`
+				} `json:"node"`
+			} `json:"router"`
+		} `json:"authority"`
+	} `json:"config"`
+}
+
+type t128Config struct {
+	Config struct {
+		Authority struct {
+			Router []struct {
+				Node []struct {
+					DeviceInterface []lteDeviceInterface
+				} `json:"node"`
+			} `json:"router"`
+		} `json:"authority"`
+	} `json:"config"`
+}
+
+type lteDeviceInterface struct {
+	Name            string `json:"name"`
+	Type            string `json:"type"`
+	TargetInterface string `json:"targetInterface"`
+}
+
+type lteDeviceState struct {
+	LTE struct {
+		Active3GPPProfile struct {
+			APN string `json:"APN"`
+		} `json:"Active 3GPP Profile"`
+		SignalStrength     string `json:"Signal Strength"`
+		RSRQSignalStrength string `json:"RSRQ Signal Strength"`
+		RSSISignalStrength string `json:"RSSI Signal Strength"`
+		RadioInterface     string `json:"Radio Interface"`
+		SNRSignalStrength  string `json:"SNR Signal Strength"`
+		RSRPSignalStrength string `json:"RSRP Signal Strength"`
+		RegistrationStatus string `json:"Registration Status"`
+		ActiveBandClass    string `json:"Active Band Class"`
+		Carrier            string `json:"Carrier"`
+		ConnectionStatus   string `json:"Connection Status"`
+		Failure            struct {
+			Reason string `json:"reason"`
+		} `json:"failure"`
+	} `json:"LTE"`
+}
+
+type lteDeviceInfo struct {
+	DeviceInfo struct {
+		IMEI  string `json:"imei"`
+		ICCID string `json:"iccid"`
+		IMSI  string `json:"imsi"`
+	} `json:"Device Info"`
+}
+
+// T128Lte is an input for metrics of a 128T router instance
+type T128Lte struct {
+	CollectorName  string `toml:"collector_name"`
+	RouterName     string `toml:"router_name"`
+	NodeName       string `toml:"node_name"`
```

Comment:
Public (capitalized) and `toml:"..."` mean users can specify these in the telegraf config. It seems like all except the `CollectorName` should be private instead of public.